version: "3.9"

networks:
  proxy:
  internal:

volumes:
  caddy_data:
  caddy_config:

services:
  caddy:
    container_name: proxy-discovery
    build:
      context: .
      dockerfile: ./Caddy-AutoProxy-Dockerfile
    image: saile2204/caddy-proxy:latest
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
      - "2019:2019"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - caddy_data:/data
      - caddy_config:/config
    networks:
      - proxy
    cgroup: host
    deploy:
      labels: # Global options
        caddy.email: eliasstepanik@web.de
  
  server:
    container_name: proxy-server
    build:
      context: .
      dockerfile: ./AutoProxy.Server/Dockerfile
    image: saile2204/autoproxy-server:latest
    tty: true
    restart: unless-stopped
    networks:
      - internal
  
  ddns-updater:
    container_name: proxy-ddns
    build:
      context: .
      dockerfile: ./AutoProxy.DdnsUpdater/Dockerfile
    image: saile2204/autoproxy-ddns-updater:latest
    environment:
      GRPC_HOST: "http://proxy-server:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    tty: true
    restart: unless-stopped
    depends_on:
      - server
    networks:
      - internal
  
  port-manager:
    container_name: proxy-ports
    build:
      context: .
      dockerfile: ./AutoProxy.PortManager/Dockerfile
    image: saile2204/autoproxy-portsync:latest
    environment:
      GRPC_HOST: "http://proxy-server:8080"
      PROXMOX_FIREWALL_GROUP: "dynamic_portainer"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    tty: true
    restart: unless-stopped
    depends_on:
      - server
    networks:
      - internal
    


